packages/headless/src/modules/global-search/services/elastic.service.ts

```diff
@@ -0,0 +1,178 @@
+import { HttpStatus, Injectable } from "@nestjs/common";
+import { ElasticsearchService } from "@nestjs/elasticsearch";
+import { searchConfig } from "config/search";
+import { IProductSearchResponse } from "models";
+import { ProductSrarchDatabase } from "src/database/mongodb/search";
+import { Product } from "src/entity/product";
+import { errorResponse, successResponse } from "src/utils/response";
+import { IServiceResponse } from "src/utils/response/service.response.interface"; 
+import { ISearchProductResponse } from "../rest/dto";
+import { productSearchSchema } from "./product.schema";
+
+@Injectable()
+export class ElasticService {
+  constructor(private readonly esService: ElasticsearchService, private readonly productSearchDB: ProductSrarchDatabase) {}
+
+
+  public async createIndex() {
```

Comment:
where you are using this function?

Reply:
This will be usefull when admin needs to update search database using apis.

Reply:
run this on server startup. just create indexes and mappings